docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,489 @@
+
+    ![Add Azure credentials](./media/actions/deploy/azure-credentials.jpg)
+    **Figure 2**: Add Azure credentials.
+
+1. You'll consume this secret in a workflow in later steps. To access it, use the variable notation `${{}}`. In this case, `${{ AZURE_CREDENTIAL }}` will be populated with the JSON you saved.
+
+## Add environments
+
+[Environments](https://docs.github.com/actions/reference/environments) are used as a *logical boundary*. You can add approvals to environments to ensure quality. You can also track deployments to environments and specify environment-specific values (secrets) for configuration.
+
+For this example, you're going to split the actual Azure environment into two *logical* environments called `PRE-PROD` and `PROD`. When you deploy the web app, you'll deploy to the staging slot of the Azure web app, represented by the `PRE-PROD` environment. When you're ready to deploy to `PROD`, you'll just perform a slot swap.
+
+In this case, the only difference between the environments is the slot that you're deploying to. In real life, there would typically be different web apps (and separate web app plans), separate resource groups, and even separate subscriptions. Typically, there's an SPN per environment. You may want to override the `AZURE_CREDENTIAL` value that you saved as a repository secret by creating it as an *environment secret*.
+
+> [!NOTE]
+> Precedence works from Environment to repository. If a targeted environment has a secret called `MY_SECRET`, then that value is used. If not, the repository value of `MY_SECRET` (if any) is used.
+
+1. Select **Settings** and then **Environments** in your repository. Select **New Environment**:
+
+    ![Create an environment](./media/actions/deploy/new-env.jpg)
+    **Figure 3**: Create an environment.
+
+1. Enter `PRE-PROD` and select **Configure environment**:
+
+    ![Name the environment](./media/actions/deploy/pre-prod-env.jpg)
+    **Figure 4**: Name the environment.
+
+1. Since deploying to a staging slot doesn't affect the web app, you can safely deploy to the slot without requiring an approval first. A reviewer could be added if desired. For this example, leave the Environment protection rules empty.
+
+> [!NOTE]
+> If you target an environment in a workflow and it does not exist, an "empty" environment is created automatically. The environment would look exactly the same as the `PRE-PROD` environment - it would exist, but would not have any protection rules enabled.
+
+1. Select **Environments** again and again select **New Environment**. Now enter `PROD` as the name and select **Configure environment**.
+
+1. Check the **Required reviewers** rule and add yourself as a reviewer. Don't forget to select **Save protection rules**:
```

Comment:
This note without being indented will break the numbering.

Reply:
Correct. Fixed.